### ISSUE TITLE
[regression] Menu search no longer matches within words

### ISSUE THREAD

**Issue by lucb1e** [linuxmint/cinnamon#9444]
**Issue**

After upgrading to Debian Bullseye, the `menu@cinnamon.org` applet no longer matches text within a word, you now have to type the beginning of a word. This makes it harder to find nicely typable matches for software you want to launch frequently.

**Steps to reproduce**

1. Search for, for example, `idu` (for me that lies very nicely under my hands' natural position)
2. Observe that there are no search results

**Expected behaviour**

It should match the screenshotting tool because its description contains the word "individual". This is only entry that contains that word, so another app can't sort above it and be launched by accident.

**Other information**

The regression was introduced in this commit: https://github.com/linuxmint/cinnamon/commit/d40ba4f900da90f09ecda78f3b139f7bef0fd4cf

**Comment by lucb1e** [comment]
If anyone wants to monkeypatch it:

```
$ sudo patch /usr/share/cinnamon/applets/menu@cinnamon.org/applet.js
2760c2760
<                 if ((Util.latinise(app.get_name().toLowerCase()).split(' ').some(word => word.startsWith(pattern))) || // match on app name
---
>                 if ((Util.latinise(app.get_name().toLowerCase()).indexOf(pattern) > -1) || // match on app name
2762c2762
<                     (app.get_description() && Util.latinise(app.get_description().toLowerCase()).split(' ').some(word => word.startsWith(pattern))) || // match on description
---
>                     (app.get_description() && Util.latinise(app.get_description().toLowerCase()).indexOf(pattern) > -1) || // match on description
```

(End with <kbd>Ctrl+D</kbd>)

**Comment by leigh123linux** [comment]
Can you reproduce it with cinnamon-4.6.x?

**Comment by lucb1e** [comment]
Yes, this is in Cinnamon `4.6.6-1` as included in Debian Bullseye.

**Comment by ItzSwirlz** [comment]
ahha. want to note real fast that uh, patchwise for Debian Buster it's not as easy to make a patch if you just give us that? Thanks for reporting-we can handcode it in but look at my patches like this muffin regression: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=965382

@leigh123linux I can PR it to save time for you and I'll later setup a patch in bullseye.

**Comment by ItzSwirlz** [comment]
Yeah and this isn't a lonely problem.
-i'll be tagging dupes of this.

**Comment by NikoKrause** [comment]
Fixed by #9598
